### PR TITLE
Moving to `clang-format-3.8` from `clang-format-3.6`.

### PR DESCRIPTION
**Pull request by dkorolev**
TO: @mzhurovich, and thanks in advance for the review.
CC @grixa for pointing out the build problem, I've fixed it.
CC @nyospe -- `clang-format-3.8` quits the nasty habit of overly indenting the lambdas, try if you like it this way.
CC @deathbaba --- recommend upgrading as well.

Thanks,
Dima

### CODE REVIEW

File: CompactTSV/test.cc [hunk 1]
```diff
@@ -116,12 +128,12 @@ TEST(CompactTSV, Smoke) {
     std::cerr << "Original TSV size:\t" << golden_size << "b, or\t" << golden_size / (1024u * 1024u) << "MB.\n";
     std::cerr << "Packed   TSV size:\t" << packed_size << "b, or\t" << packed_size / (1024u * 1024u) << "MB.\n";
     const double K = 1e-3;  // microseconds -> milliseconds.
-    std::cerr << "Generate:                                   " << K *(t_a_end - t_a_begin).count() << "ms.\n";
-    std::cerr << "Pack:                                       " << K *(t_b_end - t_b_begin).count() << "ms.\n";
-    std::cerr << "Unpack into std::ostringstream:             " << K *(t_c_end - t_c_begin).count() << "ms.\n";
-    std::cerr << "Unpack into std::string-s:                  " << K *(t_d_end - t_d_begin).count() << "ms.\n";
-    std::cerr << "Unpack into std::pair<const char*, size_t>: " << K *(t_e_end - t_e_begin).count() << "ms.\n";
-    std::cerr << "Unpack into UniqueChunk-s:                  " << K *(t_f_end - t_f_begin).count() << "ms.\n";
+    std::cerr << "Generate:                                   " << K * (t_a_end - t_a_begin).count() << "ms.\n";
```
Comment (dkorolev) [inline CompactTSV/test.cc]: Ain't this great?

File: TypeSystem/Evolution/test.cc [hunk 2]
```diff
@@ -595,346 +582,318 @@ TEST(TypeEvolutionTest, AutogeneratedCurrentStructsAreReadyForEvolution) {
...
-      from.s.x = 488885;
-      from.s.y = 488886;
-      from.s.z = "max";
-      current::type_evolution::Evolve<SchemaXX, SchemaXX::StructWithStruct>::template Go<SchemaV1>(from, into);
-      EXPECT_EQ(488885, into.s.x);
-      EXPECT_EQ(488886, into.s.y);
-      EXPECT_EQ("max", into.s.z);
-    }
-  }
+  { // Plain `SimpleStruct` tests, just copy the fields over, no evolution.
```
Comment (dkorolev) [inline TypeSystem/Evolution/test.cc]: Oh shit. Looking into.

**Comment by dkorolev** [comment]
OK, the `TypeSystem/Evolution/test.cc` f*ckup is now fixed. I hope this PR is good to merge.

Also experimenting with [more options](https://github.com/dkorolev/Current/compare/clang-format-3.8...dkorolev:clang-format-3.8-deeper?expand=1), PTAL as you have a few minutes to mediate.

Thanks,
Dima

File: Karl/schema_karl.h [hunk 3]
```diff
@@ -226,10 +226,11 @@ CURRENT_STRUCT(up) {
...
   CURRENT_DEFAULT_CONSTRUCTOR(up) {}
-  CURRENT_CONSTRUCTOR(up)(const std::chrono::microseconds start_time_epoch_microseconds,
-                          const std::string& last_keepalive_received,
-                          const std::chrono::microseconds last_keepalive_received_epoch_microseconds,
-                          const std::string& uptime)
+  CURRENT_CONSTRUCTOR(up)
+  (const std::chrono::microseconds start_time_epoch_microseconds,
+   const std::string& last_keepalive_received,
+   const std::chrono::microseconds last_keepalive_received_epoch_microseconds,
+   const std::string& uptime)
```
Comment (mzhurovich) [inline Karl/schema_karl.h]: This is not good...

Comment (dkorolev) [inline Karl/schema_karl.h]: Indeed, it's ugly. How do you feel about https://github.com/dkorolev/Current/commit/9cbf35c17d65bbec8d935ab96ff74cc3c43e8793?

I'm OK keeping the space in `if (...)`, `for (...)`, and `while (...)`, but that would also introduce them into `CURRENT_STRUCT (Foo)` and `CURRENT_CONTSTRUCTOR (Foo)`.

File: Bricks/util/test.cc [hunk 4]
```diff
@@ -785,7 +779,8 @@ TEST(ObjectInterface, Smoke) {
   }
   {
     object_with_interface_test_namespace::NamespacedExampleObjectInterface<
-        object_with_interface_test_namespace::NamespacedObjectWithInterface> o;
+        object_with_interface_test_namespace::NamespacedObjectWithInterface>
+        o;
```
Comment (mzhurovich) [inline Bricks/util/test.cc]: ))

File: Karl/schema_karl.h (hunk 3 above)
Comment (mzhurovich) [inline Karl/schema_karl.h]: I'm strongly for spaces in `if ()`, `for ()` etc.
Looks like we should admit that `clang-format` fucks up our macros in one or another way and we'll have to use `clang-format off` for them.

Comment (dkorolev) [inline Karl/schema_karl.h]: Yeah. Let's table the `clang-format` migration for now then.

**Pull request closed without merging**